homeassistant/components/shelly/config_flow.py

```diff
@@ -460,7 +472,7 @@ def async_supports_options_flow(cls, config_entry: ConfigEntry) -> bool:
         return (
             get_device_entry_gen(config_entry) in RPC_GENERATIONS
             and not config_entry.data.get(CONF_SLEEP_PERIOD)
-            and config_entry.data.get("model") != MODEL_WALL_DISPLAY
+            and cast(bool, config_entry.data.get(CONF_SCRIPT))
```

Comment:
We don't have this data for existing entries so we are going to allow enabling script for Wall display in this case which is incorrect, we probably need to keep the `and config_entry.data.get("model") != MODEL_WALL_DISPLAY` and add this condition as an additional guard for new devices

Reply:
Good point!